### DISCUSSION TITLE
Youtube ban third party apps & ads blocker

### DISCUSSION THREAD

**Discussion by Raffy-ID** [TeamNewPipe/NewPipe#10979] [Questions]
### Checklist

- [X] I made sure that there are *no existing issues or discussions* - [open](https://github.com/TeamNewPipe/NewPipe/issues) or [closed](https://github.com/TeamNewPipe/NewPipe/issues?q=is%3Aissue+is%3Aclosed) - which I could contribute my information to.
- [X] I have read the [FAQ](https://newpipe.net/FAQ/) and my question isn't listed.
- [X] I have taken the time to fill in all the required details. I understand that the question will be dismissed otherwise.
- [X] I have read and understood the [contribution guidelines](https://github.com/TeamNewPipe/NewPipe/blob/dev/.github/CONTRIBUTING.md).

### What is/are your question(s)?

Youtube has now restrict anyone that use ads blocker, now third party apps starting to face restriction due to new youtube 2024 policy, how this is impact Newpipe as Frontend alternative to Youtube?

### Additional information

https://www.theverge.com/2024/4/15/24131338/youtube-ad-blocker-crackdown-mobile-apps

**Comment by AudricV** [comment]
These changes already started since 2022 after Vanced's shutdown and blocked NewPipe and/or its extractor a few times already to access streams we get from the Android YouTube app that we try to emulate (see past releases' changelogs).

The current checks mainly concern the ability to get streams from the Android app as a third-party/modified client: YouTube requires currently a valid result from Google's DroidGuard virtual machine (used by SafetyNet and probably Google Play Integrity) and a valid YouTube visitor ID to get valid streams (this current requirement has been learned with a personal collaboration with a reVanced team member).

The extractor in the current release, 0.26.1, is affected by the DroidGuard integrity checks, as we can't get streams from the Android client anymore (but still from other clients, see TeamNewPipe/NewPipeExtractor#1168).

The YouTube app you get with reVanced patches was also affected, as it used the same workaround as us. Hopefully it can now pass the DroidGuard VM thanks to microG's GmsCore after a few changes.

[As a reminder from our FAQ entry](https://newpipe.net/FAQ/#download-youtube-api), we do not use YouTube official APIs but only unofficial/internal ones and HTML + JavaScript resources. YouTube only talks about its official APIs in its community announcement.

Only the future will tell us how this "crackdown" will affect NewPipe.

**Reply by mikeloeven** [reply]
Might want to look into teaming up with the Ublock Origin devs so far they have had very good luck countering everything you tube throws at them they probably have a lot of experience that can assist with keeping new pipe functional

**Reply by AudricV** [reply]
The anti-adblock checks on the desktop website are completely different from what is required to get streams currently.
